watertap/unit_models/crystallizer.py

```diff
@@ -316,6 +329,20 @@ def eq_mass_balance_constraints(b, j):
                 if (p, j) in b.properties_solids[0].phase_component_set
             )
 
+        @self.Constraint()
+        def eq_pure_vapor_flow_rate(b):
+            return (
+                b.properties_pure_water[0].flow_mass_phase_comp["Vap", "H2O"]
+                == b.properties_vapor[0].flow_mass_phase_comp["Vap", "H2O"]
+            )
```

Comment:
It's not immediately clear why we need this additional stateblock for water vapor. Furthermore, we are creating a constraint that sets the the new stateblock's mass flowrate to that of `properties_vapor`, which I worry could potentially lead to degeneracy in the model. 

If you are simply trying to use an alias for water in vapor phase, you could alternatively try adding a reference without creating a whole new stateblock. I might be missing something important here that would justify doing it this way but curious to hear your thoughts.

Reply:
Ah I think you saw my response to Mayo's similar question and hope it helps clarify. Please let me know if you have further questions regarding this new property package.

Reply:
I did, but it didn't justify the need to make a new stateblock, which is still foggy to me. I saw your explanation for the energy from vapor calculation, but I don't think you need a new stateblock for that either.

Reply:
Please let me put it in this way, the existing state `properties_vapor` describes a superheated state of the vapor (for example 0.35 bar, 80 C in effect 1 ), and I need a state describing the condensation state of vapor at the same pressure(0.35 bar, 72 C), in order to consider the sensible heat there. It's a virtual state because it's not a real state from the crystallizer but happens in a condenser or heat exchanger. 

So 80 C is in equilibrium with the brine feed `properties_out` as calculated by the model, while 72 C should be calculated from the equilibrium with pure water, which is not described in the current model.

Perhaps there's a way to derive the actual condensation temperature? I couldn't find a way with the existing states so I created a new one.  This new property should probably be renamed to `properties_vapor_equil` to indicate the equilibrium state.

Reply:
OK, that makes a lot more sense now. Thanks for the detailed response!